### PR TITLE
moved BFH Library and added "Library" suffix

### PR DESCRIPTION
**Pull request by R3dst0ne**
I decided to do a PR with the two approved things from #670.

### CODE REVIEW

**Comment by BFH-ktt1** [comment]
For me this is okay.

**Comment by MarcinOrlowski** [comment]
@R3dst0ne can you please fix the conflicts?

**Comment by MarcinOrlowski** [comment]
Please hold syncing until #713 landed

**Comment by R3dst0ne** [comment]
I now get the File Error `The built-in library Wiring is not available in this version` while starting. If I click OK it works just fine and the Wiring library is loaded and fully functional. I don't know where this error comes from. Do you have any ideas?

**Comment by MarcinOrlowski** [comment]
I think this is the culprit:

    public static final String _ID = "WiringLibrary";

you should not change _ID as this is not a class name (nor mapped to). Restore "Wiring" and try.

**Comment by R3dst0ne** [comment]
I changed all the `_ID`s in the *Library files. ^^'
_Wiring_ is the only one that makes problems.

**Comment by MarcinOrlowski** [comment]
But you should not change `_ID`s in the first place as you will not be able to load any project as this ID is used to reference the library. There's huge warning comment next to each `_ID` in the source :). As why is Wiring causing you troubles after the change - there're still some hardcoded "Wiring" in various places (i.e. in loaders) so once you changed the `_ID` of the library, it stopped matching. Please restore the `_ID`s.

**Comment by R3dst0ne** [comment]
@MarcinOrlowski Thank you for the support.

@BFH-ktt1 This should be ready to be merged now.

**Comment by MarcinOrlowski** [comment]
Side note, what really happened here:

> I now get the File Error The built-in library Wiring is not available in this version while starting. If I click OK it works just fine and the Wiring library is loaded and fully functional.

So all worked correctly. And no, it's not a "Wiring" library that is loaded and shown. It's "WiringLibrary" that is. "Wiring" was not found, hence the error message :) Once you changed the `_ID` and then tried to load old project, it refered to `Wiring` library But there was no such library any more (because you changed the `_ID`) so error popped up. And load continued. What you saw loaded is "Base" library and library referenced by `_ID` in `Wiring.java` (or in your case `WiringLibrary.java`) is part of base library, that it was loaded anyway. But technically it is different library. If you'd save your project, corrected `_ID` and tried to load it again, you'd see error that `WiringLIbrary` library is not found, but again, you'd see contents of Base library.

**Comment by MarcinOrlowski** [comment]
Just a note I am postponing review of this one intentionally to see the fate of #724 as if that one is gonna land, then it'd be far easier to merge this one after than vice versa.

**Comment by BFH-ktt1** [comment]
@MarcinOrlowski : Should I merge #724 before this one or just in order of the PR's?

**Comment by MarcinOrlowski** [comment]
@BFH-ktt1 I think #724 should land first.

**Comment by BFH-ktt1** [comment]
@MarcinOrlowski , @R3dst0ne : Could you look into the conflicts?

File: src/main/java/com/cburch/logisim/std/bfh/BfhLibrary.java
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/bfh/BfhLibrary.java:45]: Value of library `_ID` needs to stay unaltered despite the refactoring.

File: src/main/java/com/cburch/logisim/std/hdl/HdlLibrary.java
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/hdl/HdlLibrary.java:47]: Value of library `_ID` needs to stay unaltered despite the refactoring.

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]
Please restore specified `_ID`s as per code comments.

File: src/main/java/com/cburch/logisim/std/bfh/BfhLibrary.java
Comment (R3dst0ne) [reply src/main/java/com/cburch/logisim/std/bfh/BfhLibrary.java]: My bad, i missed those to e8902a4.

**Review by MarcinOrlowski — APPROVED** [review]
LGTM

**Pull request merged**
